drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1619,13 +1619,15 @@ int usb_dc_ep_flush(const u8_t ep)
 	return 0;
 }
 
-int usb_dc_ep_write(const u8_t ep, const u8_t *const data,
-		    const u32_t data_len, u32_t *const ret_bytes)
+static int usbd_ep_write(const u8_t ep, const u8_t *const data,
+			 const u32_t data_len, u32_t *const ret_bytes,
+			 bool zlp)
 {
 	LOG_DBG("ep_write: ep %d, len %d", ep, data_len);
 	struct nrf_usbd_ctx *ctx = get_usbd_ctx();
 	struct nrf_usbd_ep_ctx *ep_ctx;
 	u32_t bytes_to_copy;
+	u8_t flags = zlp ? NRFX_USBD_TRANSFER_ZLP_FLAG : 0;
```

Comment:
Cannot you use here something like?

```
flags = (ep == 0x80) && (data_len == 64) NRFX_USBD_TRANSFER_ZLP_FLAG : 0;
```

Reply:
No, because Windows expects a transfer without ZLP when `wLength == 64` and transfer with ZLP when `wLength > 64`.

Reply:
So in this case length is 64, so we do not send ZLP?

Reply:
OK, Windows asks length=255 despite we return 64 in the previous call, and it needs to know that this is the last packet, this can be the general case then.

Reply:
Basically this is what Windows expects:

| Descriptor length | wLength given by host |        ZLP?        |
|:-----------------:|:---------------------:|:------------------:|
|         64        |           64          |         :x:        |
|         64        |          255          | :heavy_check_mark: |